src/qililab/automatic_calibration/calibration_node.py

```diff
@@ -0,0 +1,690 @@
+
+        Returns:
+            dict: outputs of the executed notebook file.
+
+        Raises:
+            IncorrectCalibrationOutput: In case no outputs, incorrect outputs or multiple outputs where found. Incorrect outputs are those that do not contain `check_parameters` or is empty.
+        """
+        # Parsing file
+        outputs_lines: list[str] = []
+        with open(f"{self.nb_folder}/{file_name}") as file:  # pylint: disable=unspecified-encoding
+            lines = file.readlines()
+            outputs_lines.extend(line for line in lines if line.find(logger_output_start) != -1)
+
+        # Check how many lines contain the outputs, to raise the corresponding errors:
+        if not outputs_lines:
+            logger.error(
+                "Aborting execution. No output found, check the automatic-calibration output cell is implemented in %s",
+                self.nb_path,
+            )
+            raise IncorrectCalibrationOutput(f"No output found, check automatic-calibration notebook in {self.nb_path}")
+
+        if len(outputs_lines) > 1:
+            logger.error(
+                "Aborting execution. More than one output found, please output the results once in %s",
+                self.nb_path,
+            )
+            raise IncorrectCalibrationOutput(f"More than one output found in {self.nb_path}")
+
+        # When only one line of outputs, use that one:
+        return self._from_logger_string_to_output_dict(outputs_lines[0], self.nb_path)
+
+    def _from_logger_string_to_output_dict(self, logger_string, input_path):
+        """Returns the output dictionary from the logger string, or raises errors if ``logger_string`` doesn't follow the expected format.
+
+        Args
+            logger_string (str): The logger string containing the output dictionary to extract.
+            input_path (str): Path of the notebook that generated the output, to raise errors.
+
+        Returns:
+            dict: Kwargs for the output parameters of the notebook.
+
+        Raises:
+            IncorrectCalibrationOutput: In case no outputs, incorrect outputs or multiple outputs where found. Incorrect outputs are those that do not contain `check_parameters` or is empty.
+        """
+        logger_splitted = logger_string.split(logger_output_start)
+
+        # In case something unexpected happened with the output we raise an error
+        if len(logger_splitted) < 2:
+            logger.error(
+                "Aborting execution. No output found, check the automatic-calibration output cell is implemented in %s",
+                input_path,
+            )
+            raise IncorrectCalibrationOutput(f"No output found, check automatic-calibration notebook in {input_path}")
+        if len(logger_splitted) > 2:
+            logger.error(
+                "Aborting execution. More than one output found, please output the results once in %s",
+                input_path,
```

Comment:
We could merge the lines, since we don't need the temporary variable.


```suggestion
        clean_data = logger_splitted[1].split("\\n")[0].replace('\\"', '"').split("\n")[0]
        out_dict = json.loads(clean_data)
```

Reply:
Hmm, the reason they are separated now is because the first line is working for taking into account other encodings we could have received, so its like a very separated work, so I dunno really 🙃

Reply:
So, if you don't see it being bad, since I like ti separated, I'll close this one! Thanks!